#Sprint 2

CharlieMcFly opened the pull request:
Ajout du scan des beacons environnantes
Association des beacons environnantes avec le produit scanner.
Envoie de la liste des beacons environnantes.

Oupsla committed:
feat(beacon): working with eddystone and altbeacon

Oupsla committed:
feat(beacon): scan with estimote scanner

Oupsla committed:
feat(beacon): adding distance computation

CharlieMcFly committed:
feat(postProduct): Changement du get en post

modification de l'appel à l'api en post
récupération des beacons
envoie de la position au serveur (préparation)

denishamann committed:
feat(postProduct): Trilateration

CharlieMcFly committed:
feat(getBeacons): Récupération des beacons

CharlieMcFly committed:
Merge branch 'feat/mapping' of https://github.com/Oupsla/BtrShop-android into feat/mapping

# Conflicts:
#	app/build.gradle
#	app/src/main/java/io/btrshop/products/ProductsPresenter.java
#	app/src/main/java/io/btrshop/products/domain/model/BeaconObject.java
#	app/src/main/java/io/btrshop/products/domain/model/Position.java

CharlieMcFly committed:
feat(objectMapped) : modification des beacons object et de la position

denishamann committed:
feat(mapping): trilateration update

CharlieMcFly committed:
feat(objectMapped) : trilateration fonctionnel + probleme position

CharlieMcFly committed:
feat(objectMapped) : mise à jour de la position du produit

suppression de la trilatération et délégation au serveur
envoie du post avec liste de beacon et distance
get du produit

CharlieMcFly committed:
fix(url) : mise a jour de l'url de test vers prod

CharlieMcFly committed:
fix(pullrequest) : suppress trilateration + minsdk 18

Commit:
Merge pull request #6 from Oupsla/feat/mapping

Feat/mapping

CharlieMcFly committed:
fix(travis) : ajout du travis_wait pour l'install dans le yml travis

CharlieMcFly committed:
core(travis) : ajout du travis_wait pour l'install dans le yml travis

CharlieMcFly committed:
core(travis) : ajout du travis_wait 20 pour l'install dans le yml travis

duribreux commented on .travis.yml:
```diff
@@ -11,6 +11,7 @@ android:
     - extra-google-google_play_services
-
+install :
+  - travis_wait 20
```
Is this really necessary ? What's the long task you're checking every 20 seconds ? (If I understand correctly)

CharlieMcFly replied:
The process to compile everything on the project take a lot of time and make travis fail. To resolve this problem i used this solution at the link : 
https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

duribreux commented on app/src/main/java/io/btrshop/BtrShopApplication.java:
```diff
@@ -24,6 +28,10 @@ public void onCreate() {
                 .apiModule(new ApiModule(mContext.getResources().getString(R.string.base_url_api)))
                 .build();
+
+        // TODO: put your App ID and App Token here
```
Is this still up-to-date ? There already are an App ID & App Token line 34

duribreux commented on app/src/main/java/io/btrshop/data/source/api/ProductsService.java:
```diff
@@ -16,13 +19,17 @@
 
     /**
-     *  Function that retrieve the call for the product with EAN.
+     * Envoie de la position de l'article et recupération de celui ci
```
Keep the code and documentation in english please

duribreux commented on app/src/main/java/io/btrshop/data/source/api/ProductsService.java:
```diff
@@ -16,13 +19,17 @@
-     * @param ean, of the product
-     * @return the product
+     * @param ean
+     * @param position
```
Improve a bit this documentation, as far as I can see it's a List of position and a link to this class could be nice.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsActivity.java:
```diff
@@ -2,75 +2,104 @@
+public class ProductsActivity extends AppCompatActivity implements ProductsContract.View {
 
-public class ProductsActivity extends AppCompatActivity implements ProductsContract.View{
+    // Constantes
```
English.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsActivity.java:
```diff
@@ -79,15 +108,78 @@ public void onClick(View v) {
 
+        // Permissions and bluetooth
+        verifyBluetooth();
+        checkAndRequestPermissions();
```
You removed bluetooth from the manifest and you request it there. Is this normal behavior ?

duribreux commented on app/src/main/java/io/btrshop/products/ProductsActivity.java:
```diff
@@ -130,15 +222,15 @@ public IdlingResource getCountingIdlingResource() {
                 String result = data.getStringExtra("result");
                 Log.d("SCAN", result);
                 dialog = new MaterialDialog.Builder(this)
                         .title("Récupération produit")
```
Avoid string contants like that. Prefer using the values/strings.xml file and use english by default if you don't setup the values-fr/strings.xml

duribreux commented on app/src/main/java/io/btrshop/products/ProductsActivity.java:
```diff
@@ -150,22 +242,67 @@ public void showScan() {
+                throw new RuntimeException("Bluetooth LE not supported by this device");
+            }
+            if (!getApplicationContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)) {
+                throw new RuntimeException("Bluetooth LE not supported by this device");
```
Use strings.xml file

duribreux commented on app/src/main/java/io/btrshop/products/ProductsActivity.java:
```diff
@@ -150,22 +242,67 @@ public void showScan() {
+            if (bluetoothError) {
+                dialog = new MaterialDialog.Builder(ProductsActivity.this)
+                        .title("Bluetooth not enabled")
+                        .content("Please enable bluetooth in settings and restart this application.")
```
I won't put this comment again but just put all your messages in the strings.xml file

duribreux commented on app/src/main/java/io/btrshop/products/ProductsBeacon.java:
```diff
@@ -0,0 +1,121 @@
+
+    private Context pbContext;
+
+    private final double COEFF1 = 0.42093;
```
Some documentation of those values would be awesome

duribreux commented on app/src/main/java/io/btrshop/products/ProductsBeacon.java:
```diff
@@ -0,0 +1,121 @@
+        beaconManager.setEddystoneListener(new BeaconManager.EddystoneListener() {
+            @Override
+            public void onEddystonesFound(List<Eddystone> list) {
+
```
Nothing ? That's a bit odd to scan for no reason

CharlieMcFly replied:
Was use for the eddystone mode of the beacon, we don't need this anymore.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsBeacon.java:
```diff
@@ -0,0 +1,121 @@
+
+    public double calculateDistance(int txPower, double rssi) {
+        if (rssi == 0) {
+            return -1.0; // if we cannot determine accuracy, return -1.
```
Throw an error and handle it properly instead of some error value.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsBeacon.java:
```diff
@@ -0,0 +1,121 @@
+            return -1.0; // if we cannot determine accuracy, return -1.
+        }
+
+        double ratio = rssi*1.0/txPower;
```
What's the meaning of this ratio and all. Maybe some comments would be nice.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsPresenter.java:
```diff
@@ -29,37 +32,64 @@ public ProductsPresenter(Retrofit retrofit, ProductsContract.View mView) {
 
-                });
 
+        if(listBeacon != null) {
```
beacons.getListBeacons() should just return an empty list so you won't have to use this kind of condition.

duribreux commented on app/src/main/java/io/btrshop/products/ProductsPresenter.java:
```diff
@@ -29,37 +32,64 @@ public ProductsPresenter(Retrofit retrofit, ProductsContract.View mView) {
+                    .subscribe(new Observer<Product>() {
+
+                        @Override
+                        public void onCompleted() {
```
If you don't have to override remove unused code

CharlieMcFly replied:
It's an "implements", i have to put this function or it doesn't work.

duribreux commented on app/src/main/java/io/btrshop/products/domain/model/BeaconJson.java:
```diff
@@ -0,0 +1,42 @@
+    public BeaconJson(String uuid, double distance) {
+        this.uuid = uuid;
+        this.dist = distance/1000;
+
```
empty line

duribreux commented on app/src/main/res/values/strings.xml:
```diff
@@ -62,5 +62,6 @@
     <string name="info_product_model">Model</string>
     <string name="title_activity_details_product">Details Product</string>
     <string name="base_url_api">https://btrshopcloud.herokuapp.com/</string>
+    <string name="base_url_test">http://10.42.0.249:8080/</string>
```
Might be cleaner to handle this through gradle file

duribreux commented on app/src/main/java/io/btrshop/BtrShopApplication.java:
```diff
@@ -24,6 +28,10 @@ public void onCreate() {
+
+        // TODO: put your App ID and App Token here
+        // You can get them by adding your app on https://cloud.estimote.com/#/apps
+        EstimoteSDK.initialize(getApplicationContext(), "hamann-denis-gmail-com-s-y-nqc", "2c217585b54fa97b46a39f1b9f3d603f");
```
We really don't want to put this on github, it's private imo. Handle it through gradle property or something like that (Not familiar with this might need some research)

CharlieMcFly replied:
I use this : https://github.com/codepath/android_guides/wiki/Storing-Secret-Keys-in-Android
to put my secret api and token in a secret ressources file, which is in my .gitignore

duribreux commented on build.gradle:
```diff
@@ -25,13 +25,13 @@ task clean(type: Delete) {
 ext {
     // Sdk and tools
-    minSdkVersion = 18
+    minSdkVersion = 23
```
Woow. Who is responsible for such huge change ? Beacons lib ?

aveuiller commented on build.gradle:
```diff
@@ -25,13 +25,13 @@ task clean(type: Delete) {
+    minSdkVersion = 23
     targetSdkVersion = 24
     compileSdkVersion = 24
     buildToolsVersion = '24.0.2'
```
It's better to keep `targetSdkVersion`, `compileSdkVersion`, `buildToolsVersion`, and `supportLibraryVersion` to the same major release, i.e. 25 here.

CharlieMcFly committed:
chore: change/remove line for the pull request :

 hide api key and api token in resources : secret.xml
 change some comment in english
 put some string in resources file
 change manifest to have bluetooth autorisation
 remove empty method
 explain how we use calculteDistance for beacon
 add all sdk to 25

CharlieMcFly committed:
chore : fix fail travis by removing the sdkeddystone we don't use anymore

CharlieMcFly committed:
chore (travis) : sdk 25 upgrade

CharlieMcFly committed:
chore (travis) : bug travis exited with 137

CharlieMcFly committed:
chore (readme) : update readme

CharlieMcFly committed:
chore (travis) : update travis

CharlieMcFly committed:
chore (travis) : update travis

CharlieMcFly committed:
chore (travis) : update travis

CharlieMcFly committed:
chore (travis) : update travis

CharlieMcFly committed:
chore (travis) : update travis
